gui/static/js/components/gridview.js

```diff
@@ -212,63 +199,58 @@ export default class GridView extends React.Component {
         const scale = this.state.scale;
         const xOffset = parseInt(this.state.xOffset);
         const yOffset = parseInt(this.state.yOffset);
+        const x_int = this.state.x_int;
+        const y_int = this.state.y_int;
 
         for(var i=0; i<40; i=i+1){
+            //horizontal lines
             lines_y[i] = new PIXI.Graphics();
             lines_y[i].lineStyle(1, 0x0000FF, 1);
 
-            lines_y[i].moveTo(0,i*65*scale/100);
-            lines_y[i].lineTo(VIEW_WIDTH,i*65*scale/100);
-            lines_y[i].x = 0;
-            lines_y[i].y =(i-20)*65*scale/100 + yOffset;
+            lines_y[i].moveTo(0,VIEW_WIDTH-(i-20)*y_int+yOffset);
+            lines_y[i].lineTo(VIEW_WIDTH,VIEW_WIDTH-(i-20)*y_int+yOffset);
 
             this.state.gridContainer.addChild(lines_y[i]);
 
+            //vertical lines
             lines_x[i] = new PIXI.Graphics();
             lines_x[i].lineStyle(1, 0x0000FF, 1);
 
-            lines_x[i].moveTo(i*65*scale/100,0);
-            lines_x[i].lineTo(i*65*scale/100,VIEW_WIDTH);
-            lines_x[i].x = (i-20)*65*scale/100 + xOffset;
-            lines_x[i].y = 0;
+            lines_x[i].moveTo((i-20)*x_int+xOffset,0);
+            lines_x[i].lineTo((i-20)*x_int+xOffset,VIEW_WIDTH);
             this.state.gridContainer.addChild(lines_x[i]);
         }
     }
 
     /**
      * Pseudo-constructor for a bot object.
      *
-     * @param {int} x The x coordinate of new bot.
-     * @param {int} y The y coordinate of new bot.
-     * @param {int} angle The angle of new bot, in radians.
+     * @param {float} x The x coordinate of new bot.
+     * @param {float} y The y coordinate of new bot.
+     * @param {float} angle The angle of new bot, in radians.
      * @param {string} id The id for the new bot.
-     * @param {int} size The size of the new bot, in meters
+     * @param {float} size The size of the new bot, in meters
      * @return {Object} Object which contains information about
      *     the virtual bot.
     */
-    newBot(x, y, angle, id, size) {
+    newBot(x, y, angle, id, size, type) {
         // TODO: Create a Bot class.
         var bot = {
             x: x,
             y: y,
             angle: angle,
             id: id,
-            size: size
+            size: size,
+            type: type
         };
-
-        if (size==0.15) {
-            bot.type = 'bot';
-        } else {
-            bot.type = 'scenario_obj';
-        }
         return bot;
     }
 
     /**
-     * Displays all bots onto the GridView given an array of bots.
+     * Displays all objects being tracked by the gridview.
      **/
-    displayBots() {
-        const botArray = this.state.bots;
+    displayObjects() {
```

Comment:
hi i tested addbot and the simulated bot gets drawn as a scenario object instead of a bot because the type is undefined. Info about type is lost when the code run this displayObjects()